src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -16,6 +16,291 @@ public enum UpdateStatus
     public class Utils
     {
         /// <summary>
+        /// Builds the dependencies within the list of graphNodes
+        /// </summary>
+        /// <param name="graphNodeScopeToCheck"></param>
+        public static void BuildGraphNodeDependencies(List<AssociativeGraph.GraphNode> graphNodesInScope)
+        {
+            if (graphNodesInScope == null)
+            {
+                return;
+            }
+
+            // Get the current graphnode to check against the list
+            //  [a = 10]  -> this one
+            //  c = 1
+            //  b = a
+            //  d = a
+            for (int i = 0; i < graphNodesInScope.Count; ++i)
+            {
+                AssociativeGraph.GraphNode currentNode = graphNodesInScope[i];
+                if (!currentNode.isActive)
+                {
+                    continue;
+                }
+                // Get the graphnode to check if it depends on nodeToCheckAgainstList
+                //  a = 10 (currentnode)
+                //  c = 1  (gnode) -> this is checked if it depends on [a]. If it does, add it to the dependency list of  a = 10 (currentnode)
+                //  b = a  (gnode) -> next
+                //  d = a  (gnode) -> next
+                for (int j = 0; j < graphNodesInScope.Count; ++j)
```

Comment:
It would be a potential performance issue. Is it possible to put symbols in a map so that we could get the graph nodes from a symbol in O(1)?